PyInstaller/building/build_main.py

```diff
@@ -603,6 +604,30 @@ def _check_python_library(self, binaries):
         binaries.append((os.path.basename(python_lib), python_lib, 'BINARY'))
         logger.info('Using Python library %s', python_lib)
 
+    def exclude_system_libraries(self, list_of_exceptions=[]):
+        """
+        This method may be optionally called from the spec file to exclude
+        any system libraries from the list of binaries other than those
+        containing the shell-style wildcards in list_of_exceptions.
+        Those that match '*python*' or are stored under 'lib-dynload' are
+        always treated as exceptions and not excluded.
+        """
+        def _include_binary(binary_tuple, exceptions):
+            dest = binary_tuple[0]
+            if fnmatch.fnmatch(dest, '*python*'):
+                return True
+            if dest.startswith('lib-dynload'):
+                return True
+            src = binary_tuple[1]
+            if not src.startswith('/lib') and not src.startswith('/usr/lib'):
+                return True
+            for exception in exceptions:
+                if fnmatch.fnmatch(dest, exception):
+                    return True
+            return False
+
+        self.binaries = [i for i in self.binaries if _include_binary(i, list_of_exceptions)]
```

Comment:
Same feedback [as before](https://github.com/pyinstaller/pyinstaller/pull/6016#discussion_r670641757). Put unit tests in tests/unit/test_building_utils.py.